### ISSUE TITLE
Replace `on-demand-retries` XHR ping with websocket

### ISSUE THREAD

**Issue by rauchg** [zeit/next.js#4495]
During development, the `on-demand-retries` ping causes a lot of entries in the devtools, which make debugging a bit more annoying than it should be.

Switching to WebSocket would give us a nice performance boost as well.

https://github.com/zeit/next.js/issues/1364

**Comment by jaygurnani** [comment]
This is not only an issue for chromedev tools, but this is an issue for a path traversal security attack. 

> A path traversal attack (also known as directory traversal) aims to access files and directories that are stored outside the web root folder. By manipulating variables that reference files with “dot-dot-slash (../)” sequences and its variations or by using absolute file paths, it may be possible to access arbitrary files and directories stored on file system including application source code or configuration and critical system files. It should be noted that access to files is limited by system operational access control (such as in the case of locked or in-use files on the Microsoft Windows operating system).

Here are the instructions to reproduce the attack:

1. The page parameter for the [domain]/_next/on-demand-entries-ping endpoint, allows the user to provide different paths and list existent directories in the web application.

2. For those directories that do not exist in the portal, the application returned in the response the following JSON object:

HTTP/1.1 200 OK Content-Type: application/json Content-Length: 16 X-Powered-By: Express Access-Control-Allow-Origin: [domain] Access-Control-Allow-Methods: OPTIONS, GET Date: Wed, 14 Nov 2018 19:09:56 GMT Connection: close X-N: S
{"invalid":true}

Notwithstanding, when the directory provided did exist in the platform, the response returned by the server was the following one:

HTTP/1.1 200 OK Content-Type: application/json Content-Length: 16 X-Powered-By: Express Access-Control-Allow-Origin: [domain] Access-Control-Allow-Methods: OPTIONS, GET Date: Wed, 14 Nov 2018 18:21:21 GMT Connection: close 
Connection: close X-N: S
{"success":true}

An attacker could obtain benefit of this behaviour and perform a brute force attack against the endpoint, retrieving different directories available in the platform.

**Comment by jaygurnani** [comment]
@jeffrafter You are correct, the endpoint is only available in development mode and not in production - it is not an urgent issue, but would be good still to fix.
